packages/time-series/lib/commands/index.ts

```diff
@@ -137,28 +141,75 @@ export function pushLabelsArgument(args: Array<RedisArgument>, labels?: Labels)
   return args;
 }
 
-export type SampleRawReply = {
-  2: TuplesReply<[timestamp: NumberReply, value: BlobStringReply]>;
-  3: TuplesReply<[timestamp: NumberReply, value: DoubleReply]>;
-};
+export type SampleRawReply = TuplesReply<[timestamp: NumberReply, value: DoubleReply]>;
 
 export const transformSampleReply = {
-  2(reply: SampleRawReply[2]) {
-    const [timestamp, value] = reply as unknown as UnwrapReply<typeof reply>;
+  2(reply: Resp2Reply<SampleRawReply>) {
+    const [ timestamp, value ] = reply as unknown as UnwrapReply<typeof reply>;
     return {
       timestamp,
       value: Number(value)
     };
   },
-  3(reply: SampleRawReply[3]) {
-    const [timestamp, value] = reply as unknown as UnwrapReply<typeof reply>;
+  3(reply: SampleRawReply) {
+    const [ timestamp, value ] = reply as unknown as UnwrapReply<typeof reply>;
     return {
       timestamp,
       value
     };
   }
 };
 
+export type SamplesRawReply = ArrayReply<SampleRawReply>;
+
+export const transformSamplesReply = {
+  2(reply: Resp2Reply<SamplesRawReply>) {
+    return (reply as unknown as UnwrapReply<typeof reply>)
+      .map(sample => transformSampleReply[2](sample));
+  },
+  3(reply: SamplesRawReply) {
+    return (reply as unknown as UnwrapReply<typeof reply>)
+      .map(sample => transformSampleReply[3](sample));  }
+};
+
+export function transformLablesReply2(r: RawLabels2): Labels {
+  const labels: Labels = {};
+
+  const reply = r as unknown as UnwrapReply<typeof r>;
+
+  for (const t of reply) {
+    const [k, v] = t as unknown as UnwrapReply<typeof t>;
+    const key = k as unknown as UnwrapReply<BlobStringReply>;
+    const value = v as unknown as UnwrapReply<BlobStringReply>;
+
+    labels[key.toString()] = value.toString()
```

Comment:
`value.toString()`? why?

Reply:
In V4, it was string->string

https://github.com/redis/node-redis/blob/2fc79bdfb375602e2aaba15962f57c88d8afe46b/packages/time-series/lib/commands/index.ts#L185C1-L199C2

so the Q is, do we want to require labels still be simple strings (and therefore, not dependent on the BlobString type mapping, i.e. force to string).  Or in V5 do we want to say, no, we can support BlobString values for labels.